FIX(Revision): "New" badge appears in entity deletion revision

akashgp09 opened the pull request:
<!--
Before making a pull request, please:
1. Read the guidelines for contributing
2. Verify that your changes match our coding style
3. Fill out the requested information
-->

### Problem
<!-- What are you trying to solve? -->
**This PR Fixes: [BB-523](https://tickets.metabrainz.org/projects/BB/issues/BB-523?filter=allopenissues)**


### Solution
<!-- What does this PR do to fix the problem? -->
Refactored the `isNew` property logic in **server/routes/revision.js** by taking advantage of `dataId`( which is **null** for a deleted entity)
 
### Before

![Screenshot from 2021-04-18 21-39-58](https://user-images.githubusercontent.com/55311336/115152536-25361180-a08f-11eb-84e4-6b3d5d054058.png)


### After
![Screenshot from 2021-04-18 21-39-19](https://user-images.githubusercontent.com/55311336/115152554-33842d80-a08f-11eb-9076-5de42a2c31f5.png)



### Areas of Impact
<!-- What parts of the codebase and which behaviors are affected? -->
server/routes/revision.js

MonkeyDo commented on src/server/routes/revision.js:
```diff
@@ -168,17 +168,24 @@ function diffRevisionsWithParents(orm, entityRevisions, entityType) {
+						return makePromiseFromObject({
+							changes: revision.diff(parent),
+							entity: revision.related('entity'),
+							entityAlias: revision.get('dataId') ?
```
Since we've already extracted this as a constant:

```suggestion
							entityAlias: dataId ?
```

akashgp09 replied:
> Great, that's a good improvement!
> 
> I think it would be great to take that opportunity to mark deleted entities with a `Deleted` badge with a red background, now that you've got that piece of info in your logic.
> I guess the cleanest would be to pass a new `isDeletion` property.
> 
> What do you think?

MonkeyDo reviewed (commented):
Great, that's a good improvement!

I think it would be great to take that opportunity to mark deleted entities with a `Deleted` badge with a red background, now that you've got that piece of info in your logic.
I guess the cleanest would be to pass a new `isDeletion` property.

What do you think?

akashgp09 commented:
Added the deleted badge to mark deleted entitites. Let me know if any changes are required :)

![Screenshot from 2021-04-19 19-08-31](https://user-images.githubusercontent.com/55311336/115247363-5ae90200-a144-11eb-903c-eb968a2a7da1.png)

MonkeyDo commented on src/server/routes/revision.js:
```diff
@@ -168,17 +168,27 @@ function diffRevisionsWithParents(orm, entityRevisions, entityType) {
+						let isNew = false;
+						let isDeletion = false;
+						if (!parent) {
+							isNew = dataId;
```
```suggestion
							isNew = Boolean(dataId);
```

I guess this is a bit clearer and makes sure we use a boolean.

MonkeyDo commented on src/server/routes/revision.js:
```diff
@@ -168,17 +168,27 @@ function diffRevisionsWithParents(orm, entityRevisions, entityType) {
+						let isDeletion = false;
+						if (!parent) {
+							isNew = dataId;
+							isDeletion = !dataId;
```
I think this could be moved above to the variable definition:
`const isDeletion = !dataId;`

MonkeyDo reviewed (changes requested):
Yes, that's nice, thanks for adding that :)

MonkeyDo reviewed (commented):
OK, this is starting to look really nice!
I went and deployed this PR to test.bookbraine.org so that we can do some testing on a common space.
It's working pretty well and showing the right badges where expected most of the time, but I did find a couple of issues:

If the entity has been merged into another entity, it will display the "deleted" badge, because for merges we also remove the dataId of any merged entity (apart from the one we are merging into).
I think it should be relatively easy to make another "merged" badge (and perhaps that will entirely replace the current way we display merged entities in a revision. That's for another PR)
An example of that: https://test.bookbrainz.org/revision/29798

In another case, I deleted an entity that had some previous revisions, and it appears with the "new" badge instead of "deleted". Not sure what is happening here:
https://test.bookbrainz.org/revision/29849
I deleted the entity `6ef597ce-9f15-4c91-ae8d-83c18a0c6797` and nothing else in that revision and the database seems to reflect that correctly:
![image](https://user-images.githubusercontent.com/6179856/115271601-caa7be80-a13d-11eb-8945-002f0799e178.png)

akashgp09 commented:
> I think it should be relatively easy to make another "merged" badge (and perhaps that will entirely replace the current way we display merged entities in a revision. That's for another PR)

Ok sure will open a PR for a`merged`badge to mark merged entities once this PR gets merged.

> In another case, I deleted an entity that had some previous revisions, and it appears with the "new" badge instead of "deleted". Not sure what is happening here:
https://test.bookbrainz.org/revision/29849
I deleted the entity 6ef597ce-9f15-4c91-ae8d-83c18a0c6797 and nothing else in that revision and the database seems to reflect that correctly.

Yeah thanks for reporting it. I have made required changes and it seems to work fine now

![Screenshot from 2021-04-20 08-37-29](https://user-images.githubusercontent.com/55311336/115331522-b1d9f000-a1b3-11eb-9fe5-1d82a820cef3.png)

MonkeyDo reviewed (approved):
All working fine now, thank you ! :)
